### ISSUE TITLE
Value Converter Wizard Parameter Type is ignored

### ISSUE THREAD

**Issue by lachlanwgordon** [mfractor/mfractor-feedback#153]
When using the Value converter, setting the parameter type has no affect on the generate code. I'm guessing the intended behaviour is that is would do a cast and type check similar to what you do with the input.

e.g. I'm making a value converter from int to double to that should multiply the int by a factor specified in XAML.

I enter double in the Parameter Type field
![Screen Shot 2019-12-05 at 12 10 31 pm](https://user-images.githubusercontent.com/29908924/70194984-6ce92100-1758-11ea-9f86-edd5286b22b3.png)

The generated code has no reference to the parameter type
```
[ValueConversion(typeof(int), typeof(double))]
    public class StringLengthToLabelWidth : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
        {
            if (value is int == false)
            {
                return default(double);
            }

            var input = (int)value;

            // TODO: Put your value conversion logic here.

            return default(double);
        }

        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
        {
            throw new NotImplementedException();
        }
    }
```

I would have expected something like this to be added with the type used as parameter, just before the TODO: Comment
```
if (parameter is double == false)
            {
                return default(double);
            }

            var param = (double)parameter;
```

**Comment by lachlanwgordon** [comment]
Forgot to mention:
Visual Studio Community 2019 for Mac (Preview) Version 8.4 Preview (8.4 build 2535)
MFractor 3.9.0

MFractor 3.9.0 doesn't officially target 8.4 but if you switch to stable VS, install mfractor, then switch back to preview it works.

**Comment by matthewrdev** [comment]
@lachlanwgordon I'll update the code template to include:

         var param = (double)parameter;

Would this be suitable?

**Comment by matthewrdev** [comment]
Changed the template to:

using System;
using System.Globalization;
using Xamarin.Forms;

namespace $namespace$
{
    [$value_conversion_attribute$(typeof($input_type$), typeof($output_type$))]
    public class $name$ : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
        {
            if (value is $input_type$ == false)
            {
                return default($output_type$);
            }

            var input = ($input_type$)value;
            var param = ($parameter_type$)parameter;

            // TODO: Put your value conversion logic here.

            return default($output_type$);
        }

        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
        {
            throw new NotImplementedException();
        }
    }
}
```

**Comment by lachlanwgordon** [comment]
That will do the trick although I think adding type checking would be better to prevent an Invalid Cast Exception and return default value for output type if parameter is wrong.

That kind of comes down to a style preference as to whether you'd prefer to have an exception or return null/default if you give it a bad parameter which is a decision each user will have to make.

Given there's no way to catch exceptions in code that's triggered by xaml I tend toward return default here as the default behaviour for generated code.

I'm happy either way.

**Comment by matthewrdev** [comment]
You can always remove the parameter line immediately after creating the snippet. The wizard will get you most of the way there and you can fill in the final blanks :)

**Comment by matthewrdev** [comment]
![image](https://user-images.githubusercontent.com/3260473/70197387-cb65cd80-175f-11ea-8491-c5106793146e.png)

**Comment by lachlanwgordon** [comment]
Beautiful.

**Comment by matthewrdev** [comment]
I'll release this shortly.

**Comment by matthewrdev** [comment]
@lachlanwgordon This has been fixed and will be released in MFractor 3.9.1.
